solutions/silver/cf-104468H.mdx

```diff
@@ -0,0 +1,189 @@
+---
+id: cf-104468H
+source: CF
+title: Ammar-utiful Array
+author: Kevin Sheng
+---
+
+<Spoiler title="Hint 1">
+
+The problem mentions prefixes of arrays; that should ring a couple bells in your head.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+If a color's prefix of length $10$ isn't Ammar-utiful, will a prefix of $11$ ever be Ammar-utiful?
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+## Explanation
+
+First, we compute prefix sums for each color so we can figure out in $\mathcal{O}(1)$ time what all the prefixes for each color are.
+
+The queries of type $1$ are a bit trickier to handle because they add an amount to every element that's *not* a certain color.
+Because of this, simply keeping track of how much has been added to each color would be too slow.
+
+To work around this, we instead keep track of how much has *not* been added to each color along with a counter for the total amount added.
+
+For example, say we have the colors $1$, $2$, and $3$, along with the following queries:
+```
+1 1 2
+1 2 3
+```
+Our total amount added after this would be $5$.
+$2$ would have not been added to the color $1$, and $3$ would not have been added to the color $2$.
+
+To get how much was added for a specific color, we take the total counter and subtract how much was *not* added to that specific color.
+
+After all this, we can do a binary search on the maximum prefix for each query of type $2$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$ for preprocessing and $\mathcal{O}(\log N)$ for each query
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <map>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+int main() {
+	std::ios::sync_with_stdio(false);
+	std::cin.tie(NULL);
```

Comment:
u need this right

Reply:
i would never put it in if it wasn't needed